plugins/modules/net_tools/nmcli.py

```diff
@@ -2100,8 +2100,12 @@ def _compare_conn_params(self, conn_info, options):
                 if key == self.mtu_setting and self.mtu is None:
                     self.mtu = 0
                 if key == 'vpn.data':
-                    current_value = list(map(str.strip, current_value.split(',')))
-                    value = list(map(str.strip, value.split(',')))
+                    current_values = []
+                    for item in current_value.split(','):
+                        name, value_ = map(str.strip, item.split('=', 1))
```

Comment:
Better use list comprehension instead of `map()`.
```suggestion
                        name, value_ = [part.strip() for part in item.split('=', 1)]
```
Unfortunately, this has a problem: what happens if there is no `=` in there?

Reply:
I made it more flexible~